eclair-core/src/main/scala/fr/acinq/eclair/io/Peer.scala

```diff
@@ -239,6 +234,12 @@ class Peer(val nodeParams: NodeParams, remoteNodeId: PublicKey, switchboard: Act
     case Event(_: Channel.OutgoingMessage, _) => stay // we got disconnected or reconnected and this message was for the previous connection
   }
 
+  private val reconnectionTask = context.actorOf(ReconnectionTask.props(nodeParams, remoteNodeId), "reconnection-task")
+
+  onTransition {
+    case _ -> DISCONNECTED => reconnectionTask ! Peer.Transition(stateData, nextStateData)
```

Comment:
But we're also interested in transition to the `CONNECTED` state, aren't we?

Reply:
Dammit. This hasn't been caught by unit tests because we are mocking the peer with proper behavior.

Reply:
Fixed with 9c6eed3.